src/gtk/webview_webkit2.cpp

```diff
@@ -38,6 +39,20 @@
 #include <JavaScriptCore/JSValueRef.h>
 #include <JavaScriptCore/JSStringRef.h>
 
+#if WEBKIT_CHECK_VERSION(2, 10, 0)
+#define wxHAVE_WEBKIT_WEBSITE_DATA_MANAGER
+#endif
+
+// Function to check webkit version at runtime
+bool wx_check_webkit_version(int major, int minor, int micro)
+{
+    const unsigned int version = webkit_get_major_version() * 10000 +
+                                 webkit_get_minor_version() * 100 +
+                                 webkit_get_micro_version();
+    const unsigned int required = major * 10000 + minor * 100 + micro;
+    return version >= required;
+}
```

Comment:
Can `wx_check_webkit_version` please be moved into a separate commit, and perhaps be merged separately ASAP?  I could use it for the other webkit2 changes.

Reply:
I've added it as a separate commit. But this PR can probably be merged soon anyway and would include this change.

Reply:
Thanks - that will make it easier to backport - since all of the changes in this PR will probably not be backportable.